tests/visualization_tests/matplotlib_tests/test_contour.py

```diff
@@ -288,3 +289,91 @@ def test_contour_subplots_have_correct_axis_labels_and_ranges() -> None:
         xlims = [ax.get_xlim() for ax in subplots[:, index]]
         assert all_equal(xlims)
         assert all(range_covers(param_range_with_padding, xlim) for xlim in xlims)
+
+
+@pytest.mark.parametrize("value", [float("inf"), -float("inf")])
+def test_nonfinite_removed(recwarn: WarningsRecorder, value: float) -> None:
+
+    study = create_study()
+    study.add_trial(
+        create_trial(
+            value=0.0,
+            params={"param_a": 1.0, "param_b": 0.0},
+            distributions={
+                "param_a": FloatDistribution(0.0, 1.0),
+                "param_b": FloatDistribution(0.0, 1.0),
+            },
+        )
+    )
+    study.add_trial(
+        create_trial(
+            value=1.0,
+            params={"param_a": 0.0, "param_b": 10.0},
+            distributions={
+                "param_a": FloatDistribution(0.0, 1.0),
+                "param_b": FloatDistribution(0.0, 10.0),
+            },
+        )
+    )
+    study.add_trial(
+        create_trial(
+            value=value,
+            params={"param_a": 0.0, "param_b": 1.0},
+            distributions={
+                "param_a": FloatDistribution(0.0, 1.0),
+                "param_b": FloatDistribution(0.0, 1.0),
+            },
+        )
+    )
+
+    # To check if contour lines have been rendered (meaning +-inf trials were removed),
+    # we should be looking if artists responsible for drawing them are preset in the final plot.
+    # Turns out it's difficult to do reliably (No information which artists are responsible for
+    # drawing contours) so instead we are checking for warning raised by matplotlib
+    # when contour plot fails. TODO(xadrianzetx) Find a better way to test this.
+    plot_contour(study)
+    for record in recwarn.list:
+        assert "No contour levels were found within the data range" not in str(record.message)
+
+
+@pytest.mark.parametrize(
+    "objective,value",
+    [(0, float("inf")), (0, -float("inf")), (1, float("inf")), (1, -float("inf"))],
+)
```

Comment:
nit
```suggestion
@pytest.mark.parametrize("objective", (0, 1))
@pytest.mark.parametrize("value", (float("inf"), -float("inf")))
```

Reply:
Done in 26305b3e43d13008afeedaaa153360657d463975.